messaging/broker.go

```diff
@@ -818,6 +871,49 @@ func (t *Topic) WriteMessage(m *Message) error {
 	return nil
 }
 
+// Truncate attempts to delete topic segments such that the total size of the topic on-disk
+// is equal to or less-than maxSize. Returns the number of bytes deleted, and error if any.
+// This function is not guaranteed to be performant.
+func (t *Topic) Truncate(maxSize int64) (nBytesDeleted int64, err error) {
+	t.mu.Lock()
+	defer t.mu.Unlock()
+
+	var segments Segments
+	segments, err = ReadSegments(t.Path())
+	if err != nil {
+		return
+	}
+
+	var size int64
+	for {
+		size, err = segments.Size()
+		if err != nil {
+			return
+		}
+		if size <= maxSize {
+			return
+		}
+
+		if len(segments) < 2 {
+			// Always leave 1 segment around, for current writes.
+			return
+		}
+
+		s := segments[0]
+		size, err = s.Size()
+		if err != nil {
+			return
+		}
+		if err = os.Remove(s.Path); err != nil {
+			return
+		}
+		nBytesDeleted += size
+		segments = segments[1 : len(segments)-1]
```

Comment:
I think this is slicing an item off on both ends.  Shouldn't it be `segments[1:]`?